catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/SourceOperations.java

```diff
@@ -0,0 +1,335 @@
+                // only add the local catalogProviderdescriptor
+                addCatalogSourceDescriptor(sourceDescriptors);
+            }
+
+            response = new SourceInfoResponseImpl(sourceInfoRequest, null, sourceDescriptors);
+
+        } catch (RuntimeException re) {
+            LOGGER.warn("Exception during runtime while performing getSourceInfo: {}",
+                    re.getMessage());
+            LOGGER.debug("Exception during runtime while performing getSourceInfo", re);
+            throw new SourceUnavailableException(
+                    "Exception during runtime while performing getSourceInfo");
+
+        }
+
+        return response;
+    }
+
+    //
+    // Helper methods - check visibility
+    //
+
+    /**
+     * Checks that the specified source is valid and available.
+     *
+     * @param source the {@link Source} to check availability of
+     * @return true if the {@link Source} is available, false otherwise
+     */
+    boolean sourceIsAvailable(Source source) {
+        if (source == null) {
+            LOGGER.warn("source is null, therefore not available");
+            return false;
+        }
+        try {
+            if (LOGGER.isDebugEnabled()) {
+                LOGGER.debug("Checking if source \"{}\" is available...", source.getId());
+            }
+
+            // source is considered available unless we have checked and seen otherwise
+            boolean available = true;
+            Source cachedSource = frameworkProperties.getSourcePoller()
+                    .getCachedSource(source);
+            if (cachedSource != null) {
+                available = cachedSource.isAvailable();
+            }
+
+            if (!available) {
+                LOGGER.warn("source \"{}\" is not available", source.getId());
+            }
+            return available;
+        } catch (ServiceUnavailableException e) {
+            LOGGER.warn("Caught ServiceUnavaiableException", e);
+            return false;
+        } catch (Exception e) {
+            LOGGER.warn("Caught Exception", e);
+            return false;
+        }
+    }
+
+    /**
+     * Retrieves the {@link SourceDescriptor} info for all {@link FederatedSource}s in the fanout
+     * configuration, but the all of the source info, e.g., content types, for all of the available
+     * {@link FederatedSource}s is packed into one {@link SourceDescriptor} for the
+     * fanout configuration with the fanout's site name in it. This keeps the individual
+     * {@link FederatedSource}s' source info hidden from the external client.
+     */
+    private SourceInfoResponse getFanoutSourceInfo(SourceInfoRequest sourceInfoRequest)
+            throws SourceUnavailableException {
+
+        SourceInfoResponse response;
+        SourceDescriptorImpl sourceDescriptor;
+        try {
+
+            // request
+            if (sourceInfoRequest == null) {
+                throw new IllegalArgumentException("SourceInfoRequest was null");
```

Comment:
Exception is being used for flow control, granted its on an error path but it could be cleaner.